docs/contributing/999-rfcs/2024-02-15-resource-pinning.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Resource Pinning
+description: Protection of resources from accidental deletion/moves
+---
+
+# Resource Pinning
+
+- **Author(s):**: @MarkMcCulloh
+- **Submission Date**: 2024-02-15
+- **Stage**: Proposal
+- **Stage Date**: 2024-02-15
+
+Introduces `pin` as a mechanism to protect resources from accidental deletion/moves in the logical application tree.
+
```

Comment:
Please add a "Requirements" section at the top with a simple list of scenarios/use-cases/problems this RFC is trying to solve. It will help us verify that the solution addresses all the scenarios.

(there is a lot of info in the original issue that you can compile into this list).